libc/spec/posix.td

```diff
@@ -1632,6 +1632,17 @@ def POSIX : StandardSpec<"POSIX"> {
                 ArgSpec<SizeTType>,
                 ArgSpec<LSearchCompareT>
             ]
+        >,  
+        FunctionSpec<
+            "lsearch",
+            RetValSpec<VoidPtr>,
+            [
+                ArgSpec<ConstVoidPtr>,
+                ArgSpec<VoidPtr>,
+                ArgSpec<SizeTPtr>,
+                ArgSpec<SizeTType>,
+                ArgSpec<LSearchCompareT>
+            ]
```

Comment:
Note: old hdrgen is about to be deleted in https://github.com/llvm/llvm-project/pull/117220, so landing this may conflict with that.